lib/mapper.js

```diff
@@ -60,41 +60,40 @@ exports.page = function(page){
  * @api private
  */
 
-exports.track = function(track) {
+exports.track = function(track, settings) {
   // If the event name contains "unsubscribe", instead send an unsubscribe payload
   if (/[Uu]nsubscribe/.test(track.event())) {
     return reject({
-      auth_token: this.settings.authToken,
+      auth_token: settings.authToken,
       id: track.userId(),
       extras: {
         created_at: time(track.timestamp()),
         source: 'segment'
       }
     });
+  } 
+  var identity; 
+  if (typeof track.proxy('context.traits.email') !== 'undefined') {
```

Comment:
I believe we want to explicitly only check `context.traits.email` here. This was not written by me.